Packages/Fluxity/Tests/DispatcherTests.cs

```diff
@@ -139,12 +139,79 @@ public void Dispatch_WhenMultipleMatchingReducersAndEffectsRegistered_ShouldInvo
     }
 
     [Test]
-    public void Dispatch_WhenReducerDispatches_ShouldThrow()
+    public void Dispatch_WhenReducerDispatches_ShouldDispatchTwice()
     {
+        var count = 0;
         _feature.SetState(new DummyState());
         _staticDispatcher = _dispatcher;
+        //this is a hack that we don't actually want to allow but now we queue so we can ensure the following thing happens
         var reducer = new PureFunctionReducerBinder<DummyState, DummyCommand>(DispatchDuringReduce);
         _feature.Register(reducer);
+        _dispatcher.OnDispatch += _ => count++;
+
+        _dispatcher.Dispatch(new DummyCommand());
+
+        Assert.AreEqual(2, count);
+    }
+
+    [Test]
+    public void Dispatch_WhenEffectDispatchesSameButLimited_ShouldDispatchTwiceAndEffectInvokeTwice()
+    {
+        var count = 0;
+        _feature.SetState(new DummyState());
+        _staticDispatcher = _dispatcher;
+        var dummyEffect = new DummyEffect();
+        dummyEffect.Action = () =>
+        {
+            if (count < 2)
+                _staticDispatcher.Dispatch(new DummyCommand());
+        };
+        var effect = new EffectBinding<DummyCommand>(dummyEffect.DoEffect);
+        _dispatcher.RegisterEffect(effect);
+        _dispatcher.OnDispatch += _ => count++;
+
+        _dispatcher.Dispatch(new DummyCommand());
+
+        Assert.AreEqual(2, count);
+        Assert.AreEqual(2, dummyEffect.callCount);
+    }
+
+    [Test]
```

Comment:
I'd add an additional test the same as this one, except it dispatches two or more different commands within the `dummyEffect.Action`, and that it indeed does dispatch all of them, and in the correct order.